### PR TITLE
[Caffe2] Add moments op in caffe2

### PR DESCRIPTION
**Pull request by xiaomengy**

### CODE REVIEW

File: caffe2/operators/moments_op.cc [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+  const int dY_size = std::accumulate(
+      dY_dims.cbegin(), dY_dims.cend(), 1, std::multiplies<int>());
+  const int dX_size = std::accumulate(
+      dX_dims.cbegin(), dX_dims.cend(), 1, std::multiplies<int>());
+  const int ndim = dX_dims.size();
+  std::vector<int> index(ndim, 0);
+  const T norm = static_cast<T>(dY_size) / static_cast<T>(dX_size);
+  for (int dX_index = 0; dX_index < dX_size; ++dX_index) {
+    const int dY_index =
+        math::internal::GetIndexFromDims(ndim, dY_dims.data(), index.data());
```
Comment (houseroad) [inline caffe2/operators/moments_op.cc]: If dY_size < dX_size, it seems we may have overflow here.

File: caffe2/operators/moments_op.cc [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+    .NumInputs(1)
+    .NumOutputs(2)
+    .SetDoc(R"DOC(
+  Computes the mean and variance of the input tensor's element along the
+  provided axes. The resulted tensor has the same rank as the input if keepdims
+  equals True.
+  If keepdims equals False, then the resulted tensor have the reduced dimension
+  pruned.
+)DOC")
+    .Arg("axes", "A list of integers, along which to reduce.")
```
Comment (houseroad) [inline caffe2/operators/moments_op.cc]: Could you add the default behavior if axes is not provided?

Comment (BIT-silence) [reply caffe2/operators/moments_op.cc]: Done.

**Review by houseroad — APPROVED** [review]

**Pull request merged**